plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestingScyllaServer.java

```diff
@@ -45,28 +51,46 @@
     private final CassandraSession session;
 
     public TestingScyllaServer()
+            throws Exception
     {
         this("2.2.0");
     }
 
     public TestingScyllaServer(String version)
+            throws Exception
     {
         container = new GenericContainer<>("scylladb/scylla:" + version)
                 .withCommand("--smp", "1") // Limit SMP to run in a machine having many cores https://github.com/scylladb/scylla/issues/5638
                 .withExposedPorts(PORT);
         container.start();
 
-        Cluster.Builder clusterBuilder = Cluster.builder()
-                .withProtocolVersion(V3)
-                .withClusterName("TestCluster")
-                .addContactPointsWithPorts(ImmutableList.of(
-                        new InetSocketAddress(container.getContainerIpAddress(), container.getMappedPort(PORT))))
-                .withMaxSchemaAgreementWaitSeconds(60);
+        ProgrammaticDriverConfigLoaderBuilder driverConfigLoaderBuilder = DriverConfigLoader.programmaticBuilder();
+        driverConfigLoaderBuilder.withString(DefaultDriverOption.PROTOCOL_VERSION, ProtocolVersion.V3.name());
+        driverConfigLoaderBuilder.withDuration(DefaultDriverOption.CONTROL_CONNECTION_AGREEMENT_TIMEOUT, java.time.Duration.ofSeconds(30));
+        // allow the retrieval of metadata for the system keyspaces
+        driverConfigLoaderBuilder.withStringList(DefaultDriverOption.METADATA_SCHEMA_REFRESHED_KEYSPACES, List.of());
+
+        CqlSessionBuilder cqlSessionBuilder = CqlSession.builder()
+                .withApplicationName("TestCluster")
+                .addContactPoint(new InetSocketAddress(this.container.getContainerIpAddress(), this.container.getMappedPort(PORT)))
+                .withLocalDatacenter("datacenter1")
+                .withConfigLoader(driverConfigLoaderBuilder.build());
 
-        session = new CassandraSession(
+        CassandraSession session = new CassandraSession(
                 JsonCodec.listJsonCodec(ExtraColumnMetadata.class),
-                new ReopeningCluster(clusterBuilder::build),
+                cqlSessionBuilder::build,
                 new Duration(1, MINUTES));
+
+        try {
+            checkConnectivity(session);
+        }
+        catch (RuntimeException e) {
+            session.close();
+            this.container.stop();
+            throw e;
+        }
+
```

Comment:
This connection check should be an independent commit and the commit body should have the motivation in my opinion.

Reply:
I think that the check is not quite necessary because it is the container's responsibility to check this aspect.